core/ir/disassemble_shared.c

```diff
@@ -1164,6 +1224,48 @@ sign_extend_immed(instr_t *instr, int srcnum, opnd_t *src)
     }
 }
 
+#define MAX_LENGTH_ALL_CATEGORIES_AS_STRING 57
+
+/* Returns a space-separated string representation of the list of categories an
+ * instruction belongs to.
+ */
+void
+get_category_names(uint category, char category_names[], size_t category_names_size)
+{
+    memset(category_names, '\0', category_names_size);
+    if (category == DR_INSTR_CATEGORY_UNCATEGORIZED) {
+        const char *category_name = instr_get_category_name(category);
+        size_t char_to_copy = strlen(category_name);
+        strncpy(category_names, category_name, char_to_copy);
```

Comment:
This is not checking category_names_size; nor is it following https://dynamorio.org/page_code_content.html#autotoc_md109 which says "Always null-terminate buffers after calling `strncpy` or `snprintf`: for statically-sized buffers, use  NULL_TERMINATE_BUFFER()."

Reply:
Using `print_to_buffer()`, which takes care of that. Done.